From c8a6c191401b637797b99eacbdc267d0ba5ca03e Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Tue, 9 Nov 2021 13:49:23 +0000
Subject: [PATCH] CMakeLists: search for Python3 Development

* fixes:
  -- Found pybind11: /usr/include (found version "2.7.0" )
  CMake Error at /jenkins/mjansa/build/ros/webos-rolling-honister/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/rclpy/3.1.0-1-r0/recipe-sysroot/usr/share/cmake/pybind11/pybind11NewTools.cmake:196 (python3_add_library):
    Unknown CMake command "python3_add_library".
  Call Stack (most recent call first):
    CMakeLists.txt:101 (pybind11_add_module)

  python3_add_library isn't defined when searching only for Interpreter:
  https://cmake.org/cmake/help/latest/module/FindPython3.html#commands

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7646b8ea..82c2333d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,7 @@ find_package(rosbag2_storage REQUIRED)
 find_package(rosbag2_transport REQUIRED)
 
 # Find python before pybind11
+find_package(Python3 COMPONENTS Development Interpreter REQUIRED)
 find_package(python_cmake_module REQUIRED)
 find_package(PythonExtra REQUIRED)
 if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
